examples/miscellaneous/plot_metadata_routing.py

```diff
@@ -177,21 +176,9 @@ def get_metadata_routing(self):
         return router
 
     def fit(self, X, y, **fit_params):
-        # `get_routing_for_object` returns a copy of the `MetadataRouter`
-        # constructed by the above `get_metadata_routing` method, that is
-        # internally called.
-        request_router = get_routing_for_object(self)
-        # Meta-estimators are responsible for validating the given metadata.
-        # `method` refers to the parent's method, i.e. `fit` in this example.
-        request_router.validate_metadata(params=fit_params, method="fit")
-        # `MetadataRouter.route_params` maps the given metadata to the metadata
-        # required by the underlying estimator based on the routing information
-        # defined by the MetadataRouter. The output of type `Bunch` has a key
-        # for each consuming object and those hold keys for their consuming
-        # methods, which then contain key for the metadata which should be
-        # routed to them.
-        routed_params = request_router.route_params(params=fit_params, caller="fit")
-
```

Comment:
I agree that directly using `process_routing` makes the developer guide easier to follow. However we should at least introduce `process_routing` and explains what it does, see below.